packages/neo-one-utils/src/changeCase.ts

```diff
@@ -0,0 +1,145 @@
+
+interface Languages {
+  readonly tr: {
+    readonly regexp: RegExp;
+    readonly map: Record<string, string>;
+  };
+  readonly az: {
+    readonly regexp: RegExp;
+    readonly map: Record<string, string>;
+  };
+  readonly lt: {
+    readonly regexp: RegExp;
+    readonly map: Record<string, string>;
+  };
+}
+
+type Locale = 'tr' | 'az' | 'lt';
+
+export const lowerCase = (str: string, locale?: Locale): string => {
+  const LANGUAGES: Languages = {
+    tr: {
+      regexp: /\u0130|\u0049|\u0049\u0307/g,
+      map: {
+        İ: '\u0069',
+        I: '\u0131',
+        İ: '\u0069',
+      },
+    },
+    az: {
+      regexp: /[\u0130]/g,
+      map: {
+        İ: '\u0069',
+        I: '\u0131',
+        İ: '\u0069',
+      },
+    },
+    lt: {
+      regexp: /[\u0049\u004A\u012E\u00CC\u00CD\u0128]/g,
+      map: {
+        I: '\u0069\u0307',
+        J: '\u006A\u0307',
+        Į: '\u012F\u0307',
+        Ì: '\u0069\u0307\u0300',
+        Í: '\u0069\u0307\u0301',
+        Ĩ: '\u0069\u0307\u0303',
+      },
+    },
+  };
+  const lang = locale === undefined ? undefined : LANGUAGES[locale];
+
+  const nextStr = String(str);
+
+  if (lang) {
+    return nextStr.replace(lang.regexp, (m) => lang.map[m]).toLowerCase();
+  }
+
+  return nextStr.toLowerCase();
+};
+
+export const upperCase = (str: string, locale?: Locale): string => {
+  const LANGUAGES: Languages = {
+    tr: {
+      regexp: /[\u0069]/g,
+      map: {
+        i: '\u0130',
+      },
+    },
+    az: {
+      regexp: /[\u0069]/g,
+      map: {
+        i: '\u0130',
+      },
+    },
+    lt: {
+      regexp: /[\u0069\u006A\u012F]\u0307|\u0069\u0307[\u0300\u0301\u0303]/g,
+      map: {
+        i̇: '\u0049',
+        j̇: '\u004A',
+        į̇: '\u012E',
+        i̇̀: '\u00CC',
+        i̇́: '\u00CD',
+        i̇̃: '\u0128',
+      },
+    },
+  };
+
+  const lang = locale === undefined ? undefined : LANGUAGES[locale];
+
+  let nextStr = String(str);
+
+  if (lang) {
+    nextStr = str.replace(lang.regexp, (m) => lang.map[m]);
+  }
+
+  return nextStr.toUpperCase();
+};
+
+export const noCase = (str: string, locale?: Locale, replacement?: string): string => {
```

Comment:
basically a nit: I see in some of the tests a lot of `(arg1, undefined, arg2)` which, from what I've gathered in our codebase, isn't a pattern we want to use. Could you refactor this as:
```ts
interface NoCaseOptions {
    readonly locale?: Locale;
    readonly replacement?: string;
}
export const noCase = (str: string, options: NoCaseOptions): string => {...}
```

this way we can avoid having `undefined` as an argument to noCase.